reader: draw last update even if we didn't draw before.

iaguis opened the pull request:
If you're doing a small read, it can happen that we reach EOF before we
draw for the first time so outputting the final draw without a prior
draw makes sense.

iaguis commented:
I don't see a reason why we should not output the final draw if we've not drawn before. Maybe I'm missing something...

mitchellh commented:
This doesn't merge cleanly anymore but I'd like to merge it, can you rebase the PR? Thanks!

iaguis committed:
reader: draw last update even if we didn't draw before.

If you're doing a small read, it can happen that we reach EOF before we
draw for the first time so outputting the final draw without a prior
draw makes sense.

iaguis commented:
Sure! Done :)
